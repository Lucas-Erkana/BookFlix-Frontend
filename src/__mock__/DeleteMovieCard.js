import React, { useState } from 'react';
import { BrowserRouter } from 'react-router-dom';

const DeleteMovieCard = () => {
  const service = [{
    name: 'Movie 1',
    image: 'https://res.cloudinary.com/dx0nauane/image/upload/v1624349663/Movie%20Card%20Images/Movie_1_1_xqjz3c.jpg',
    id: 1,
  }];

  const [showConfirmation, setShowConfirmation] = useState(false);

  const handleDeleteClick = () => {
    setShowConfirmation(false);
  };

  return (
    <BrowserRouter>
      <div className="col-md-12 d-flex flex-column flex-md-row p-2 justify-content-between align-items-center delete-card rounded">
        <div className="d-flex flex-wrap align-items-center m-1">
          <img src={service.image} alt="service" className="img-thumbnail rounded m-2 delete-img" />
          <h4>{service.name}</h4>
        </div>
        <div className="px-2">
          {showConfirmation ? (
            <div className="confirmation-message d-flex flex-column align-items-end">
              <p>
                Are you sure you want to delete the service
                {' '}
                {service.name}
                ? This action cannot be undone.
              </p>
              <div className="d-flex g-2 justify-content-end">
                <button className="btn btn-outline-danger" type="button" onClick={handleDeleteClick}>
                  Confirm Delete
                </button>
                <button className="btn btn-outline-secondary" type="button" onClick={() => setShowConfirmation(false)}>
                  Cancel
                </button>
              </div>
            </div>
          ) : (
            <button className="btn btn-danger delete-btn" type="button" onClick={() => setShowConfirmation(true)}>
              Delete
            </button>
          )}
        </div>
      </div>
    </BrowserRouter>
  );
};

export default DeleteMovieCard;
