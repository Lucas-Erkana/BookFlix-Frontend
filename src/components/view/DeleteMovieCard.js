import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { deleteMovie } from '../../redux/MoviesSlice';

const DeleteMovieCard = ({ id, name, image }) => {
  const dispatch = useDispatch();
  const [showConfirmation, setShowConfirmation] = useState(false);

  const handleDeleteClick = () => {
    dispatch(deleteMovie(id));
    setShowConfirmation(false);
  };

  return (
    <div className="col-md-12 d-flex flex-column flex-md-row p-2 justify-content-between align-items-center delete-card rounded">
      <div className="d-flex flex-wrap align-items-center m-1">
        <img src={image} alt="service" className="img-thumbnail rounded m-2 delete-img" />
        <h4>{name}</h4>
      </div>
      <div className="px-2">
        {showConfirmation ? (
          <div className="confirmation-message d-flex flex-column align-items-end">
            <p>
              Are you sure?
              This will permanently delete the service and it&apos;s reservations.
            </p>
            <div className="d-flex g-2 justify-content-end">
              <button className="btn btn-outline-danger" type="button" onClick={handleDeleteClick}>
                Confirm Delete
              </button>
              <button className="btn btn-outline-secondary" type="button" onClick={() => setShowConfirmation(false)}>
                Cancel
              </button>
            </div>
          </div>
        ) : (
          <button className="btn btn-danger delete-btn" type="button" onClick={() => setShowConfirmation(true)}>
            Delete
          </button>
        )}
      </div>
    </div>
  );
};

DeleteMovieCard.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default DeleteMovieCard;
