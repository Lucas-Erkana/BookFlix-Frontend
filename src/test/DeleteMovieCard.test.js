import { render, cleanup } from '@testing-library/react';
import DeleteMovieCard from '../__mock__/DeleteMovieCard';

afterEach(() => {
  cleanup();
});

describe('DeleteMovieCard', () => {
  it('DeleteMovieCard renders correctly', () => {
    const deleteservicecard = render(<DeleteMovieCard />);
    expect(deleteservicecard).toMatchSnapshot();
  });
});
