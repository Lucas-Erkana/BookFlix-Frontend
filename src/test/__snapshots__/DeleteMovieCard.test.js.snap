// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovieCard DeleteMovieCard renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col-md-12 d-flex flex-column flex-md-row p-2 justify-content-between align-items-center delete-card rounded"
      >
        <div
          class="d-flex flex-wrap align-items-center m-1"
        >
          <img
            alt="service"
            class="img-thumbnail rounded m-2 delete-img"
          />
          <h4 />
        </div>
        <div
          class="px-2"
        >
          <button
            class="btn btn-danger delete-btn"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col-md-12 d-flex flex-column flex-md-row p-2 justify-content-between align-items-center delete-card rounded"
    >
      <div
        class="d-flex flex-wrap align-items-center m-1"
      >
        <img
          alt="service"
          class="img-thumbnail rounded m-2 delete-img"
        />
        <h4 />
      </div>
      <div
        class="px-2"
      >
        <button
          class="btn btn-danger delete-btn"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
